src/calculator/types.ts

```diff
@@ -161,9 +171,22 @@ export namespace EAC {
 
     const OptionsDetails = Record({
         exerciseCostUSD: Number,
-        grossProceedsUSD: Number,
-        netProceedsUSD: Number,
+        grossProceedsUSD: Number.optional(),
+        netProceedsUSD: Number.optional(),
```

Comment:
Are these fields marked optional because all option transactions might have them or not? Or is the difference because of the transaction type (ExerciseAndHold or ExerciseAndSell)? If the difference is because of the transaction type I would split the OptionsDetails type to two separate types for the different transactions. The goal of these types is to describe the input data shape as accurately as possible so the compiler can catch errors better.